src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -1911,7 +1935,9 @@ private ReadOnlyCollection<TValue> GetValues()
             try
             {
                 AcquireAllLocks(ref locksAcquired);
-                List<TValue> values = new List<TValue>();
+
+                int count = GetCountInternal();
```

Comment:
I think it's worth checking count >= 0 here, as AOoRE would be confusing and OOME clear.